src/app/features/courses/components/courses-edit/courses-edit.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { CoursesService } from '../../courses.service';
+import { Course } from '../../models/course';
+
+@Component({
+  selector: 'app-courses-edit',
+  templateUrl: './courses-edit.component.html',
+  styleUrls: ['./courses-edit.component.scss'],
+})
+export class CoursesEditComponent implements OnInit {
+  private operationType: 'edit' | 'create' = 'create';
+
+  get pageTitle(): string {
+    return this.operationType === 'edit' ? 'Edit Course' : 'New Course';
+  }
+
+  public course: Course = {
+    id: Date.now(),
+    title: '',
+    description: '',
+    duration: 60,
+    creationDate: '',
+    topRated: false,
+  };
+
+  public authors: string = '';
+
+  constructor(
+    private coursesService: CoursesService,
+    private router: Router,
+    private route: ActivatedRoute
+  ) {}
+
+  ngOnInit(): void {
+    this.route.params.subscribe((params) => {
```

Comment:
Do you really need a stream subscription here? I think, you need only once in a page initialization, so you can do it in your constructor:
```
const courseId = this.route.snapshot.params.id;
.... other logic
```
In the case where you need exactly the stream - you have to unsubscribe from each subscription to avoid memory leaks in your app. It can be hard to understand from first sight since you are not familiar with rxjs, but here is in [article](https://www.digitalocean.com/community/tutorials/angular-takeuntil-rxjs-unsubscribe), how to unsubscribe properly. And another article, [Why We always should unsubscribe](https://www.learmoreseekmore.com/2021/11/why-we-have-to-unsubscribe-an-observable-in-angular-application.html#:~:text=In%20Angular%20applications%2C%20it's%20always,requests%20to%20avoid%20unwanted%20calls.).

Reply:
moved to constructor without subscription